contracts/upgrader/src/contract.rs

```diff
@@ -0,0 +1,99 @@
+use crate::error::ContractError;
+use soroban_sdk::{contract, contractimpl, Address, BytesN, Env, String, Symbol, Val};
+
+#[contract]
+pub struct Upgrader;
+
+#[contractimpl]
+impl Upgrader {
+    pub fn __constructor(_env: Env) {}
+    pub fn upgrade(
+        env: Env,
+        contract_address: Address,
+        new_version: String,
+        new_wasm_hash: BytesN<32>,
+        migration_data: soroban_sdk::Vec<Val>,
+    ) -> Result<(), ContractError> {
```

Comment:
I figured that would be part of the `upgrade` functions of the contracts, otherwise we would need to specify a unique interface that all contracts implement to make the `owner` available. That being said I still have to test if the auth call inside the upgrade function even works as expected